File: tools/hello-js/hello.js
#!/usr/bin/env node

/**
 * Hello JS - A simple greeting tool
 *
 * Usage: node hello.js [name]
 *        node hello.js --name <name>
 */

function parseArgs(args) {
  const result = {};
  for (let i = 0; i < args.length; i++) {
    if (args[i].startsWith("--") && i + 1 < args.length) {
      result[args[i].slice(2)] = args[i + 1];
      i++;
    } else if (!args[i].startsWith("--")) {
      result._positional = result._positional || [];
      result._positional.push(args[i]);
    }
  }
  return result;
}

const parsed = parseArgs(process.argv.slice(2));
const name = parsed.name || parsed._positional?.[0] || "World";

console.log(`Hello, ${name}! 👋`);

const now = new Date().toISOString();
console.log(`Generated at: ${now}`);
console.log(`Node version: ${process.version}`);

File: tools/hello-go/hello.go
package main

import (
	"fmt"
	"os"
	"runtime"
)

func main() {
	name := "World"
	args := os.Args[1:]

	for i := 0; i < len(args); i++ {
		if args[i] == "--name" && i+1 < len(args) {
			name = args[i+1]
			i++
		} else if len(args[i]) > 0 && args[i][0] != '-' {
			name = args[i]
		}
	}

	fmt.Printf("Hello, %s! 🐹\n", name)
	fmt.Printf("Go version: %s\n", runtime.Version())
}
